src/Adapter/CMS/Page/CommandHandler/DuplicateCmsPageHandler.php

```diff
@@ -0,0 +1,109 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\CMS\Page\CommandHandler;
+
+use CMS;
+use Language;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\CmsPageSettings;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\Command\DuplicateCmsPageCommand;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\CommandHandler\DuplicateCmsPageHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\ValueObject\CmsPageId;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Duplicates given cms page.
+ */
+final class DuplicateCmsPageHandler extends AbstractCmsPageHandler implements DuplicateCmsPageHandlerInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var StringModifierInterface
+     */
+    private $stringModifier;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param StringModifierInterface $stringModifier
+     */
+    public function __construct(
+      TranslatorInterface $translator,
+      StringModifierInterface $stringModifier
+    ) {
+        $this->translator = $translator;
+        $this->stringModifier = $stringModifier;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(DuplicateCmsPageCommand $command)
+    {
+        // Get current data
+        $source_cms_id = $command->getCmsPageId()->getValue();
+        $source_cms = new CMS((int) $source_cms_id);
+
+        // Prepare and save new page
+        $new_cms = new CMS();
+        $new_cms->id_cms_category = $source_cms->id_cms_category;
+        $new_cms->indexation = $source_cms->indexation;
+        $new_cms->active = false;
+        $new_cms->meta_title = $this->getNewPageTitles($source_cms->meta_title);
+        $new_cms->content = $source_cms->content;
+        $new_cms->head_seo_title = $source_cms->head_seo_title;
+        $new_cms->meta_description = $source_cms->meta_description;
+        $new_cms->meta_keywords = $source_cms->meta_keywords;
+        $new_cms->link_rewrite = $source_cms->link_rewrite;
+        $new_cms->add();
```

Comment:
so it would be like 
```suggestion
        // Get current data
        $sourceCmsId = $command->getCmsPageId()->getValue();
        $sourceCms = new CMS((int) $sourceCmsId);

        // reset the id so the new cms is saved having all the other properties copied from sourceCms
        $sourceCms->id = null;
        $sourceCms->active = false;
        $sourceCms->add();
```

Reply:
@zuk3975 Didn't go for this one, the code is now very short as I used the way of resetting ID as you proposed. :-) It's a good idea, it also duplicates some custom CMS page properties that modules could add. 👍 CLEVER BOY

Reply:
Yes, the code is short and looks good, but its not only about that. The AbstractObjectModelRepository offers some methods like addObjectModel and getObjectModel that are wrapped in a try catch, so it ensures the class is loaded correctly and throws related Core exception if something goes wrong (whereas now, it will throw legacy PrestaShop exception which is not handled in the controller). Btw it keeps things similar as in other recently migrated pages like the Product one. I would really love to see it implemented here :smile: 
If you think its too much for you, I can implement it by contributing to your branch, no problem, just let me know :v: